core-rust/state-manager/src/staging/result.rs

```diff
@@ -0,0 +1,360 @@
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use super::{ReadableLayeredTreeStore, ReadableTxAndRxTreeStore};
+use crate::accumulator_tree::storage::{ReadableAccuTreeStore, TreeSlice, WriteableAccuTreeStore};
+use crate::accumulator_tree::tree_builder::{AccuTree, Merklizable};
+use crate::{
+    AccumulatorHash, CommittedTransactionIdentifiers, EpochTransactionIdentifiers, LedgerHashes,
+    LedgerPayloadHash, LedgerTransactionReceipt, ReceiptHash, StateHash, TransactionHash,
+};
+use lazy_static::lazy_static;
+use radix_engine::state_manager::StateDiff;
+use radix_engine::transaction::{TransactionReceipt, TransactionResult};
+use radix_engine_interface::api::types::SubstateOffset;
+use radix_engine_interface::crypto::{hash, Hash};
+use radix_engine_interface::data::scrypto::scrypto_encode;
+use radix_engine_stores::hash_tree::tree_store::{
+    NodeKey, Payload, ReNodeModulePayload, ReadableTreeStore, TreeNode, WriteableTreeStore,
+};
+use radix_engine_stores::hash_tree::{put_at_next_version, SubstateHashChange};
+
+lazy_static! {
+    static ref EMPTY_STATE_DIFF: StateDiff = StateDiff::new();
+}
+
+pub struct ProcessedResult {
```

Comment:
_Note: this `ProcessedResult` is not a new thing - I moved it from `cache.rs`, because it started growing and actually cache is not interested in its internals.

In general, I also started a little refactoring here, which is supposed to split `ProcessedResult` into `Success` vs `Failure` enum - this will simplify a few callers, and allow to get rid of a few `.clone()`s. I prefer to do it in next PR._

Reply:
Ahh understood! That makes much more sense :D. Indeed we only need to care about Commits here I imagine.